docs/source/index.rst

```diff
@@ -0,0 +1,275 @@
+
+    def get_blueprint(self):
+        return plugin_blueprint
+
+So our plugin code should look like this::
+
+    from flask import render_template
+    from flask_pluginengine import Plugin, PluginBlueprint, current_plugin
+
+    plugin_blueprint = PluginBlueprint('example', __name__)
+
+
+    @plugin_blueprint.route('/plugin')
+    def hello_plugin():
+        return render_template('example_plugin:index.html', plugin=current_plugin)
+
+
+    class ExamplePlugin(Plugin):
+        """ExamplePlugin
+
+        Just an example plugin
+        """
+
+        def get_blueprint(self):
+            return plugin_blueprint
+
+
+To get more about information see `Plugin`_ API reference.
+
+Minimal Application set up
+--------------------------
+
+To create Flask application using Flask-PluginEngine we need to define it as PluginFlask application::
+
+    from flask_pluginengine import PluginFlask
+    app = PluginFlask(__name__)
+
+    @app.route("/")
+    def hello():
+        return "Hello world!"
+
+    if __name__ == "__main__":
+        app.run()
+
+Now we need to set up the configuration: namespace of plugins we will be using with the application and plugins we want to use.
+Let's use the ``example_plugin`` we created before. For example::
+
+    app.config['PLUGINENGINE_NAMESPACE'] = 'example'
+    app.config['PLUGINENGINE_PLUGINS'] = ['example_plugin']
+
+Next we need to create an instance of PluginEngine for our app and load the plugins::
+
+    plugin_engine = PluginEngine(app=app)
+    plugin_engine.load_plugins(app)
+
+Then we can access loaded plugins by calling :func:`get_active_plugins` method, it will return a dictionary.
+If we want to check if all of the plugins were loaded correclty we can also call :func:`get_failed_plugins` method that will return a dictionary with plugins that failed to load::
+
+    active_plugins = plugin_engine.get_active_plugins(app=app).values()
+
+
+Example application functionality
+---------------------------------
+
+To understand the possibilities of Flask-PluginEngine we will create a Jinja template, where we will list all the plugins we are using.
+Let's create two templates, the base one (base.html), for example::
+
+    <!DOCTYPE html>
+    <html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <title>Title</title>
+    </head>
+    <body>
+        {% block content %}
+        {% endblock %}
+    </body>
+    </html>
+
+And index.html, inheriting from the base one, for example::
+
+    {% extends 'base.html' %}
+
+    {% block content %}
+        {% for plugin in plugins %}
+            <div>{{ plugin.title }} - {{ plugin.description }}</div>
+        {% endfor %}
+    {% endblock %}
+
+And let's pass the plugins, when rendering the template in our application::
+
+    @app.route("/")
+    def hello():
+    return render_template('index.html', plugins=active_plugins)
+
+Now what we also should do is register the blueprints of all our plugins, for instance::
```

Comment:
"Now what we **should also** do is **to** register the blueprints of all our plugins..."